<?xml version="1.0"?>
<launch>
    <arg name="sim" default="True" />
    <arg name="save" default="False" />

    <rosparam command="load" file="$(find swarm_manager)/conf/swarm_conf.yaml" />

    <!-- Launch server -->
    <group unless="$(arg sim)">
        <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
        </include>
    </group>

    <!-- Launch swarm controller-->
    <node name="swarm_controller" pkg="swarm_manager" type="swarm_controller.py" output="screen">
        <param name="sim" value="$(arg sim)"/>
    </node>

    <!-- Launch formation manager-->
    <node name="formation_manager" pkg="formation_manager" type="formation_manager_ros.py" output="screen" />

    <!-- Launch trajectory planner -->
    <node name="trajectory_planner" pkg="trajectory_planner" type="trajectory_planner_ros.py" output="screen" />

    <!-- Launch flight recorder -->
    <node name="flight_recorder" pkg="swarm_manager" type="flight_recorder.py" output="screen">
        <param name="save" value="$(arg save)"/>
    </node>

    <!-- rviz -->
    <node name="rviz_markers" pkg="swarm_manager" type="rviz_markers.py" output="screen" />
    <param name="cf_description" command="$(find xacro)/xacro $(find crazyflie_description)/urdf/crazyflie.urdf.xacro" />
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find swarm_manager)/conf/crazyflie_pos.rviz" required="true" />

</launch>
